File: LongestCommonPrefix/LongestCommonPrefix.java
package LongestCommonPrefix;
// https://leetcode.com/problems/longest-common-prefix/
/*
This works by using the first word as the prefix then whenever its not equal to the next word it takes away one letter from the end until its equal then it moves onto the next word
 */
public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";
        String prefix = strs[0];

        for (int i = 1; i < strs.length; i++){
            while (strs[i].indexOf(prefix) != 0){
                prefix = prefix.substring(0,prefix.length() - 1);
            }
        }
        return prefix;
    }
}

File: Javascript/Array's-Hashing/LongestCommonPrefix.js
/**
 * https://leetcode.com/problems/longest-common-prefix/description/
 * @param {string[]} strs
 * @return {string}
 */
/* Time O(N^2) */ /* Space O(N) */
var longestCommonPrefix = function(strs) {
    var lcp = strs[0];
    for (let word of strs){
        for (var j = lcp.length - 1; j >= 0; j--){
            if (word[j] !== lcp[j]){
                lcp = lcp.slice(0,j);
            }
        }
    }

    return lcp;
};